### PR TITLE
Update ref `foreach` tests

### LINKED ISSUE #64218: Missing ref safety error inside `foreach`
```
class Program
{
    static void Main()
    {
        int i0 = 0;
        S s0 = new S(ref i0);
        {
            int i1 = 1;
            for (S s1 = new S(ref i1);;) {
                s0 = s1; // 1
                break;
            }
        }
        {
            int i1 = 1;
            foreach (S s2 in Enumerable1.Create(ref i1)) {
                s0 = s2; 
                break;
            }
        }
    }
}
ref struct S
{
    public S(ref int i) { }
}

class Enumerable1
{
    public static Enumerable1 Create(ref int p) => default;
    public Enumerator1 GetEnumerator() => default;
}

class Enumerator1
{
    public S Current => throw null;
    public bool MoveNext() => false;
}
```

Observed:
```
error CS8352: Cannot use variable 's1' in this context because it may expose referenced variables outside of their declaration scope
                s0 = s1; // 1
                     ~~
```

Expected one more error:
```
error CS8352: Cannot use variable 's2' in this context because it may expose referenced variables outside of their declaration scope
                s0 = s2; 
                     ~~
```

### PR DESCRIPTION
**Pull request by jjonescz**
Closes #64218 as suggested in https://github.com/dotnet/roslyn/issues/64218#issuecomment-1258095505.

### CODE REVIEW

File: src/Compilers/CSharp/Test/Semantic/Semantics/RefFieldTests.cs [hunk 1]
```diff
@@ -11489,6 +11489,103 @@ class Enumerator1
...
+                break;
+            }
+        }
+    }
+}
+ref struct S
+{
+    public S(ref int i) { }
+}
+ref struct Enumerable1
```
Comment (jcouv) [inline src/Compilers/CSharp/Test/Semantic/Semantics/RefFieldTests.cs:11566]: >Enumerable1 [](http://example.com/codeflow?start=11&length=11)

nit: consider naming to highlight the difference. For example, here "RefStructEnumerable"

**Review by jcouv — APPROVED** [review]
LGTM Thanks (iteration 2)

File: src/Compilers/CSharp/Test/Semantic/Semantics/RefFieldTests.cs [hunk 2]
```diff
@@ -11489,6 +11489,103 @@ class Enumerator1
...
+{
+    public static Enumerable1 Create(ref int p) => default;
```
Comment (AlekseyTs) [inline src/Compilers/CSharp/Test/Semantic/Semantics/RefFieldTests.cs:11518]: >public static Enumerable1 Create(ref int p) => default; [](http://example.com/codeflow?start=4&length=55)

There is a number of other "foreach" tests that use either `ref` or ref-struct parameter  for ```Create``` method. Should we add flavors of those where Enumerable is a ref-struct?

File: src/Compilers/CSharp/Test/Semantic/Semantics/RefFieldTests.cs (hunk 1 above)
Comment (jjonescz) [reply src/Compilers/CSharp/Test/Semantic/Semantics/RefFieldTests.cs:11566]: Good idea. Done: 066a083d27bd490a4ab1cfe94d2fb0f39ab1debd

File: src/Compilers/CSharp/Test/Semantic/Semantics/RefFieldTests.cs (hunk 2 above)
Comment (jjonescz) [reply src/Compilers/CSharp/Test/Semantic/Semantics/RefFieldTests.cs:11544]: OK, I added ref-struct flavors that I think make most sense (in particular, such tests that have an assignment inside the `foreach`): 7898375b7880b72a6ea8c477b444d41558d98599

**Comment by jjonescz** [comment]
@dotnet/roslyn-compiler ready for review

**Review by cston — APPROVED** [review]

**Comment by AlekseyTs** [comment]
This is a test-only change that can be merged with a single sign-off, we already have two.

**Pull request merged**